src/engine/TransitivePathFallback.cpp

```diff
@@ -0,0 +1,165 @@
+// Copyright 2019, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Florian Kramer (florian.kramer@neptun.uni-freiburg.de)
+
+#include "TransitivePathFallback.h"
+
+#include <memory>
+#include <optional>
+#include <utility>
+
+#include "engine/CallFixedSize.h"
+#include "engine/TransitivePathBase.h"
+#include "util/Exception.h"
+
+// _____________________________________________________________________________
+TransitivePathFallback::TransitivePathFallback(
+    QueryExecutionContext* qec, std::shared_ptr<QueryExecutionTree> child,
+    const TransitivePathSide& leftSide, const TransitivePathSide& rightSide,
+    size_t minDist, size_t maxDist)
+    : TransitivePathImpl<Map>(qec, child, leftSide, rightSide, minDist,
+                              maxDist) {}
+// _____________________________________________________________________________
+ResultTable TransitivePathFallback::computeResult() {
+  if (minDist_ == 0 && !isBoundOrId() && lhs_.isVariable() &&
+      rhs_.isVariable()) {
+    AD_THROW(
+        "This query might have to evalute the empty path, which is currently "
+        "not supported");
+  }
+  auto [startSide, targetSide] = decideDirection();
+  shared_ptr<const ResultTable> subRes = subtree_->getResult();
+
+  IdTable idTable{allocator()};
+
+  idTable.setNumColumns(getResultWidth());
+
+  size_t subWidth = subRes->idTable().numColumns();
+
+  if (startSide.isBoundVariable()) {
+    shared_ptr<const ResultTable> sideRes =
+        startSide.treeAndCol_.value().first->getResult();
+    size_t sideWidth = sideRes->idTable().numColumns();
+
+    CALL_FIXED_SIZE((std::array{resultWidth_, subWidth, sideWidth}),
+                    &TransitivePathFallback::computeTransitivePathBound, this,
+                    &idTable, subRes->idTable(), startSide, targetSide,
+                    sideRes->idTable());
+
+    return {std::move(idTable), resultSortedOn(),
+            ResultTable::getSharedLocalVocabFromNonEmptyOf(*sideRes, *subRes)};
+  }
```

Comment:
Isn't the `computeResult` method also something that can be moved into the common templated `...Impl` class to reduce the code duplication?